_posts/2017-07-15-Comments-are-on.markdown

```diff
@@ -0,0 +1,122 @@
+
+From there you need to be able to declare in a post what issue it is hooked up to. This blog uses Jekyll, which has [YAML Front Matter][yaml-fm] on posts. So, I edited our post templates to detect for a key `comment_id`.
+
+From there you need to grab the comments JSON, and move them into the DOM.
+
+I based my work on these two posts:
+
+* [GitHub hosted comments for GitHub hosted blogs][gh-2011]
+* [Replacing Disqus with Github Comments][gh-2017]
+
+However this version is more reliable (GitHub authenticated requests) and has fewer dependencies (no jQuery for example).
+
+{% raw %}
+```html
+{% if page.comment_id %}
+  <article class='post'>
+    {% include gh_comments.html %}
+  </article>
+{% endif %}
+```
+{% endraw %}
+
+This then imports the required JavaScript into the page. It feels a lot like this;
+
+{% raw %}
+
+```javascript
+var writeToComment = function(element, html) {
+  var element = document.createElement(element)
+  element.innerHTML = html
+  document.getElementById("comments").appendChild(element)
+}
+
+var loadComments = function(data) {
+  writeToComment("h2", "Comments")
+  
+  for (var i = 0; i < data.length; i++) {
+    var commentHTML = [...]
+    writeToComment("div", commentHTML)
+  }
+
+  var callToAction = [...]
+  writeToComment("div", callToAction)
+}
+
+var writeFirstComment = function() {
+  var callToAction = [...]
+  writeToComment("div", callToAction)
+}
+
+// This is mostly there now: http://caniuse.com/#feat=fetch
+if (window.fetch) {
+  var url =
+    "https://artsy-blog-gh-commentify.herokuapp.com/repos/artsy/artsy.github.io/issues/{{ page.comment_id }}/comments"
+
+  window
+    .fetch(url, { Accept: "application/vnd.github.v3.html+json" })
+    .then(function(response) {
+      return response.json()
+    })
+    .then(function(json) {
+      if(json.length) {
+        loadComments(json)
+      } else {
+        writeFirstComment()
+      }
+    })
+}
+```
+{% endraw %}
+
+No-one is going to award this JavaScript with a prize for elegance, but it works just fine. That's basically it, you can edit the DOM however you want.
+
+The full PR for these changes is here: [artsy.github.io#363][pr] - and you can see the current [HTML/JS here][current].
+
+# Why GitHub?
+
+It's easier for you to keep track of the conversations, you're likely already having a lot of conversations in a place like GitHub. This means you can use the same flow and tools as your daily job, not relying on a third party service's emails.
+
+You have good admin tools: you can edit comments, block and report problematic users. These are tools that you have for all repos.
+
+People will be using their developer accounts, which I'd like to hope they will take pride in. You're probably more likely to get high quality responses. The lack of threading is a bit of a shame in this context, but we've lived with it in GitHub Issues for this long, so I'm OK with this.
```

Comment:
You should chat with @acjay some time about UI implications of threaded comments :)